frontend/dockerfile/parser/parser.go

```diff
@@ -109,11 +109,11 @@ type directives struct {
 
 // setEscapeToken sets the default token for escaping characters in a Dockerfile.
 func (d *directives) setEscapeToken(s string) error {
-	if s != "`" && s != "\\" {
+	if s != "`" && s != `\` {
 		return errors.Errorf("invalid escape token '%s' does not match ` or \\", s)
 	}
 	d.escapeToken = rune(s[0])
-	d.lineContinuationRegex = regexp.MustCompile(`\` + s + `[ \t]*$`)
+	d.lineContinuationRegex = regexp.MustCompile(`([^\` + s + `])\` + s + `[ \t]*$|^\` + s + `[ \t]*$`)
```

Comment:
Shouldn't using optional block suffice?

```
regexp.MustCompile(`([^\` + s + `])?\` + s + `[ \t]*$`)
```

Reply:
Will have to give that a test 🤔 it's been a while since I wrote this 😂

Reply:
Ok, I recall why it was needed; the second branch is needed for line-continuation symbols on their own line, which would otherwise be handled incorrectly.

Let me add some additional comments to this block, as there's also one corner-case we can't match easily (line-continuation preceded by an escaped escape-token, e.g.;

```
RUN echo hello\\\
world
```

Which _should_ result in `echo hello\\world`, but we currently don't match